### PR TITLE
Acts as taggable

### PR DESCRIPTION
**Pull request by boomingbrake**
!!!!!!!!!!!!!!!!!! ---------- This is a prototype ---- DO NOT MERGE ------- !!!!!!!!!!!!!!!!!!!!

@iAmNathanJ 

messing with stuff ... 

we may be able to use ruby's .parameterize.underscore to replace the space with underscore.

I might of messed up something cause I just forced pushed my changes.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -2,9 +2,19 @@ class EventsController < ApplicationController
 
   def index
     @events = Event.all
+    if params[:topics] || params[:times] || params[:days] || params[:frequencies]
+      @events = Event.tagged_with(params[:topics]||params[:times] || params[:days] ||params[:frequencies], :any => true)
+    end
+    
+    @topics = ActsAsTaggableOn::Tagging.where(context: 'topics').map { |tagging| tagging.tag.name }.uniq
```
Comment (boomingbrake) [inline app/controllers/events_controller.rb]: @iAmNathanJ  we can reference these variables in the views "@topics, @days .." they should provide up with the list of unique tags for each category

File: app/views/events/_event_card.html.erb [hunk 2]
```diff
@@ -1,6 +1,5 @@
-<article class="event-card">
+<article class="event-card" data-tags="">
```
Comment (boomingbrake) [inline app/views/events/_event_card.html.erb]: might be about to utilize the  .parameterize.underscore here to populate this attribute.

File: app/views/events/_event_card.html.erb [hunk 3]
```diff
@@ -29,19 +28,21 @@
...
-          <li>Ruby</li>
-          <li>UX</li>
-          <li>C#</li>
-          <li>Unity</li>
-          <li>Friendship</li>
-          <li>Pizza</li>
-        </ul>
-      </div>
+      <ul>
+        <% event.topic_list.each do |topic| %>
```
Comment (boomingbrake) [inline app/views/events/_event_card.html.erb]: Dynamically added the card tags

File: app/views/events/_filters.html.erb [hunk 4]
```diff
@@ -1,119 +1,55 @@
...
   </svg>
 </button>
 <div id="event-filters__filters" class="event-filters__filters">
-  <div class="event-filters__heading">Filter Events by:</div>
-
-  <div class="event-filters__group" role="group" aria-labelledby="filter-group-heading-topics">
+​
+  <div class="event-filters__group" data-filter-category="topics" role="group" aria-labelledby="filter-group-heading-topics">
     <h1 class="event-filters__group-heading" id="filter-group-heading-topics">Topics:</h1>
+    <% @topics.each do |topic| %>
```
Comment (boomingbrake) [inline app/views/events/_filters.html.erb]: dynamically adding the filters based of the tags in use by events (stored in db)

File: app/views/events/_filters.html.erb [hunk 5]
```diff
@@ -1,119 +1,55 @@
...
-    </div>
-    <div class="event-filters__checkbox">
-      <input type="checkbox" id="tag-id-7" name="tag-id-7">
-      <label for="tag-id-7">Front-end</label>
-    </div>
-    <div class="event-filters__checkbox">
-      <input type="checkbox" id="tag-id-8" name="tag-id-7">
-      <label for="tag-id-8">Back-end</label>
+      <input class="event-filters__filter" type="checkbox" data-tag="<%= topic %>" id="tag-id-<%= topic %>" name="tag-id-<%= topic %>">
+      <label for="tag-id-<%= topic %>"><%= topic.parameterize.underscore %></label>
```
Comment (boomingbrake) [inline app/views/events/_filters.html.erb]: example of .parameterize.underscore

**Comment by iAmNathanJ** [comment]
@boomingbrake I don't think you messed anything up by force pushing. Once this is done, we just need to make sure all is working. We'll take a closer look before we pull the trigger.

It all looks great! The only question I have is can we, ultimately, build each category dynamically? It looks like we have to call out each category explicitly to build them and use them in the views. It's not totally necessary, but would be nice for maintenance. 

Regardless of that, another thing we can do for the view is split the filter groupings off into a partial and bring those in with each category object.

Awesome work!! I'm really excited to see this working! 👍  😄

**Comment by iAmNathanJ** [comment]
MERGED !!

**Pull request merged**